lib/sensu/client/process.rb

```diff
@@ -173,14 +173,26 @@ def execute_check_command(check)
       # VM and the EventMachine event loop, using the Sensu Extension
       # API.
       #
+      # If the check hash has an "extension" key, it will run the
+      # extension whose name that is the value of that key and 
+      # pass the check hash as an argument to safe_run(). 
+      # Otherwise, it will run the extension whose name that 
+      # is the value of the "name" key and pass nil to safe_run().
+      #
       # https://github.com/sensu/sensu-extension
       #
       # @param check [Hash]
       def run_check_extension(check)
         @logger.debug("attempting to run check extension", :check => check)
+        if check.has_key?(:extension) and @extensions[:checks].has_key?(check[:extension])
+          extension = @extensions[:checks][check[:extension]]
+          config = check
+        else
+          extension = @extensions[:checks][check[:name]]
+          config = nil
+        end
         check[:executed] = Time.now.to_i
-        extension = @extensions[:checks][check[:name]]
-        extension.safe_run do |output, status|
+        extension.safe_run(config) do |output, status|
```

Comment:
Just always pass `check` as the param, as safe_run() checks the extension arity.